### PR TITLE
[PHP 8.1] Update `mysqli_stmt::execute` description

### PR DESCRIPTION
**Pull request by saundefined**
Related [RFC](https://wiki.php.net/rfc/mysqli_bind_in_execute)

### CODE REVIEW

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 1]
```diff
@@ -142,6 +173,69 @@ mysqli_query($link, "DROP TABLE myCity");
...
 Stuttgart (DEU,Baden-Wuerttemberg)
 Bordeaux (FRA,Aquitaine)
+]]>
+   </screen>
+  </example>
+  <example>
+   <title><methodname>mysqli_stmt::execute</methodname> example as of PHP 8.1</title>
+   <para>&style.oop;</para>
+   <programlisting role="php">
+    <![CDATA[
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:183]:
```suggestion
<![CDATA[
```
Not sure if it matters, but I have seen no space before CDATA anywhere else.

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 2]
```diff
@@ -142,6 +173,69 @@ mysqli_query($link, "DROP TABLE myCity");
...
+    printf("%s (%s,%s)\n", $row[0], $row[1], $row[2]);
+}
+
+/* remove table */
+$mysqli->query('DROP TABLE myCity');
+]]>
+   </programlisting>
+   <para>&style.procedural;</para>
+   <programlisting role="php">
+    <![CDATA[
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:210]:
```suggestion
<![CDATA[
```

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 3]
```diff
@@ -142,6 +173,69 @@ mysqli_query($link, "DROP TABLE myCity");
...
+    printf("%s (%s,%s)\n", $row[0], $row[1], $row[2]);
+}
+
+/* remove table */
+mysqli_query($link, 'DROP TABLE myCity');
+]]>
+   </programlisting>
+   &examples.outputs;
+   <screen>
+    <![CDATA[
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:237]:
```suggestion
<![CDATA[
```

**Review by kamil-tekiela — COMMENTED** [review]
Thanks. I will review it later.

**Comment by kamil-tekiela** [comment]
Can you please apply similar fixes to the ones in #989?

**Comment by saundefined** [comment]
@kamil-tekiela sure, updated.

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 4]
```diff
@@ -12,12 +12,13 @@
   <para>&style.oop;</para>
   <methodsynopsis role="oop">
    <modifier>public</modifier> <type>bool</type><methodname>mysqli_stmt::execute</methodname>
-   <void/>
+   <methodparam choice="opt"><type>array</type><parameter>params</parameter><initializer>&null;</initializer></methodparam>
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:15]: It's a nullable parameter. 
```suggestion
   <methodparam choice="opt"><type class="union"><type>array</type><type>null</type></type><parameter>params</parameter><initializer>&null;</initializer></methodparam>
```

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 5]
```diff
@@ -12,12 +12,13 @@
...
   </methodsynopsis>
   <para>&style.procedural;</para>
   <methodsynopsis role="procedural">
    <type>bool</type><methodname>mysqli_stmt_execute</methodname>
    <methodparam><type>mysqli_stmt</type><parameter>statement</parameter></methodparam>
+   <methodparam choice="opt"><type>array</type><parameter>params</parameter><initializer>&null;</initializer></methodparam>
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:21]:
```suggestion
   <methodparam choice="opt"><type class="union"><type>array</type><type>null</type></type><parameter>params</parameter><initializer>&null;</initializer></methodparam>
```

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 6]
```diff
@@ -40,6 +41,14 @@
   <para>
    <variablelist>
     &mysqli.stmt.description;
+    <varlistentry>
+     <term><parameter>params</parameter></term>
+     <listitem>
+      <para>
+       An optional list &array; with variables corresponding to the parameters in the statement.
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:48]:
```suggestion
       An optional list &array; with as many elements as there are bound parameters in the SQL statement being executed. Each value is treated as a &string;.
```

File: reference/mysqli/mysqli_stmt/execute.xml [hunk 7]
```diff
@@ -126,22 +154,76 @@ $val3 = 'Aquitaine';
...
      &examples.outputs;
      <screen>
 <![CDATA[
 Stuttgart (DEU,Baden-Wuerttemberg)
 Bordeaux (FRA,Aquitaine)
+]]>
+   </screen>
+  </example>
+  <example>
+   <title><methodname>mysqli_stmt::execute</methodname> example as of PHP 8.1</title>
```
Comment (kamil-tekiela) [inline reference/mysqli/mysqli_stmt/execute.xml:174]:
```suggestion
   <title>Execute a prepared statement with an array of values</title>
```

Comment (kamil-tekiela) [reply reference/mysqli/mysqli_stmt/execute.xml:174]: You may want to change the first example's title to:
>Execute a prepared statement with bound variables

**Review by kamil-tekiela — CHANGES_REQUESTED** [review]

**Comment by saundefined** [comment]
Thank you!

**Review by kamil-tekiela — APPROVED** [review]

**Pull request merged**